cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -216,6 +216,20 @@ void CheckRoundTripResult(const std::shared_ptr<Table> expected_table,
   compute::AssertTablesEqualIgnoringOrder(merged_expected, output_table);
 }
 
+void countProjectNodeOptionsInDeclarations(const compute::Declaration& input,
+                                           int& counter) {
+  const auto& options = input.options;
+  const auto& inputs = input.inputs;
+  const auto* maybe_project_option =
+      dynamic_cast<const compute::ProjectNodeOptions*>(options.get());
+  if (maybe_project_option != NULL) {
+    counter++;
+  }
```

Comment:
```suggestion
  if (input.factory_name == "project") {
    counter++;
  }
  const auto& inputs = input.inputs;
```

Isn't it easier to just look at the name?

Reply:
Right, I missed this. Thanks for noting this one.